hack/testing-olm-upgrade/test-020-olm-upgrade-4.4-4.5.sh

```diff
@@ -0,0 +1,182 @@
+}
+
+get_es_cluster_status() {
+  oc -n openshift-operators-redhat get pods -l component=elasticsearch --no-headers=true --ignore-not-found \
+   | awk 'NR==1{print $1}' \
+   | xargs -I '{}' oc -n openshift-operators-redhat exec '{}' -c elasticsearch -- es_util --query=_cluster/health \
+   | jq .status
+}
+
+get_es_indices() {
+  oc -n openshift-operators-redhat get pods -l component=elasticsearch --no-headers=true --ignore-not-found \
+    | awk 'NR==1{print $1}' \
+    | xargs -I '{}' oc -n openshift-operators-redhat exec '{}' -c elasticsearch -- es_util --query=_cat/indices
+}
+
+read_es_indices() {
+	local -n map=$1
+	while IFS= read -r line; do
+		id="$(echo "$line" | awk '{print $3}')"
+		echo "Index: '$id' stored"
+		map[$id]=$line
+	done <<< "$(get_es_indices)"
+}
+
+compare_indices_names(){
+	local failed=false
+	local -n old_m=$1
+	local -n new_m=$2
+	for k in "${!old_m[@]}"; do
+		echo "Checking index $k..."
+		if [ ! ${new_m[$k]+_} ]; then
+			failed=true
+			printf "\t'%s' is missing in the set of indices\n" "$k"
+		else
+			printf "\t'%s' found\n" "$k"
+		fi
+	done
+	if [ "$failed" = true ]; then
+		echo "Compare indices names failed!!!"
+		return 1
+	fi
+	echo "Indices names are fully matched"
+}
+
+
+# deploy elasticsearch-operator
+log::info "Deploying elasticsearch-operator ${previous_version} from marketplace..."
+deploy_marketplace_operator "openshift-operators-redhat" "elasticsearch-operator" "$previous_version" "elasticsearch-operator"
+
+# check if the operator is running
+log::info "Verifying if elasticsearch-operator deployment is ready..."
+try_until_text "oc -n openshift-operators-redhat get deployment elasticsearch-operator -o jsonpath={.status.updatedReplicas} --ignore-not-found" "1" ${TIMEOUT_MIN}
+
+log::info "Deploying ES secrets..."
+for file in "${repo_dir}"/hack/testing-olm-upgrade/secrets/*.yaml; do
+  oc apply -f "${file}" -n openshift-operators-redhat
+done;
+
+# deploy elasticsearch CR
+log::info "Deployng ES CR..."
+oc -n "openshift-operators-redhat" create -f ${repo_dir}/hack/testing-olm-upgrade/resources/cr.yaml
+
+# check if there are elasticsearch pod
+log::info "Checking if ES pod is ready in the namespace openshift-operators-redhat..."
+try_until_text "oc -n openshift-operators-redhat get pods -l component=elasticsearch -o jsonpath={.items[0].status.phase} --ignore-not-found" "Running" ${ES_POD_TIMEOUT}
+
+log::info "Getting the previous statate of elasticsearch-operator deployment"
+# get the previous status of the elasticsearch-operator
+oc describe -n "openshift-operators-redhat" deployment/elasticsearch-operator > "$ARTIFACT_DIR"/elasticsearch-operator.describe.before_update 2>&1
+
+# check if ES cluster has 3 running pods
+log::info "Checking if the ES has 3 nodes"
+try_func_until_text get_es_pods_count "3" ${ES_POD_TIMEOUT}
+
+# check if ES cluster is in green state
+log::info "Checking if the ES cluster is all green"
+try_func_until_text get_es_cluster_status "\"green\"" ${ES_POD_TIMEOUT}
```

Comment:
do we need this to be `green`? the upgrade waits for at least `yellow`... could we wait for either value?

Reply:
Fixed by accepting 'yellow' status as well as 'green'